DateModule.js

```diff
@@ -0,0 +1,190 @@
+//coder: Katja Borchert
+
+//A Date Utility Module in the "Revealing Pattern Module" format
+/* eslint-disable no-unused-vars */
+var myObj = (function iife() {
+/* eslint-enable no-unused-vars */
+
+  //private variables:
+  var date = null;
+  
+  //private function: date instantiation
+  function _initDate(){
+    date = new Date();
+  }
+    
+  //public API:
+  
+  //setDate(): sets the internal date based on user input
+  //takes in: empty argument list, date object, or number)
+  function setDate(enteredDate) {
+  
+    //check that date is set: if not, initialize date to now
+    if (date === null) {
+      _initDate();
+    }
+	
+	/*checks the type of the parameter enteredDate:
+	check if the enteredDate is of type number
+	(excluding variables of value 'NaN')*/
+    if (typeof enteredDate === 'number' && !isNaN(enteredDate)){
```

Comment:
Using `!isNaN(enteredDate)` will work most of the time here, but for certain kinds of input, will not have the expected effect. If you pass in a boolean value, such as true or false, for example, it will do type conversion. There are a couple of other approaches you could use such as `typeof enteredDate === 'number'` or `Number.isInteger(enteredDate)` that would be more reliable.